Display "You" instead of own phone number

cascheberg opened the pull request:
### Contributor checklist

<!-- replace the empty checkboxes [ ] below with checked ones [x] accordingly -->
- [x] I am following the [Code Style Guidelines](https://github.com/WhisperSystems/Signal-Android/wiki/Code-Style-Guidelines)
- [x] I have tested my contribution
- [x] My contribution is fully baked and ready to be merged as is
- [x] I have made the choice whether I want the [BitHub reward](https://github.com/WhisperSystems/Signal-Android/wiki/BitHub-Rewards) or not by omitting or adding the word `FREEBIE` in my commit message

---
### Description

Fixes #4411

**Before:**
![GroupYouBefore](https://user-images.githubusercontent.com/3845150/71442965-a0401f80-2708-11ea-894a-9467d719bb3f.png)

**After:**
![GroupYouAfter](https://user-images.githubusercontent.com/3845150/71443009-d5e50880-2708-11ea-92a7-7bdbf0112abb.png)

// FREEBIE

RiseT commented:
"You" might be prefered to "Me":

https://material.google.com/style/writing.html#writing-language

cascheberg commented:
Updated "Me" -> "You"

RiseT commented:
You might want to change the PR title accordingly.

kevinwestern commented:
How often are PRs merged? It looks like you've sent quite a few PRs but most are still pending. I'd like to contribute, but I don't want the work to be wasted if it's never merged.

moxie0 commented on src/org/thoughtcrime/securesms/MediaPreviewActivity.java:
```diff
@@ -100,7 +100,7 @@ private void initializeActionBar() {
       relativeTimeSpan = null;
     }
-    getSupportActionBar().setTitle(recipient == null ? getString(R.string.MediaPreviewActivity_you)
+    getSupportActionBar().setTitle(recipient == null ? getString(R.string.Recipients_you)
```
what's the rationale for this change?

cascheberg replied:
It's the same string "You", so I figured it could use the same id. In the German Signal translation for instance these are translated differently, which could have been avoided.

moxie0 commented on src/org/thoughtcrime/securesms/PassphraseRequiredActionBarActivity.java:
```diff
@@ -152,9 +150,8 @@ private int getApplicationState(MasterSecret masterSecret) {
-    } else {
-      return STATE_NORMAL;
     }
+    return STATE_NORMAL;
```
why is this necessary?

cascheberg replied:
This was left over, I can revert it.

moxie0 commented on src/org/thoughtcrime/securesms/recipients/Recipients.java:
```diff
@@ -285,17 +288,18 @@ public String getSortedIdsString() {
-        fromString += ", ";
+    for (Recipient recipient : recipients) {
+      if (Util.isOwnNumber(context, recipient.getNumber())) {
+        from.push(context.getString(R.string.Recipients_you));
```
This feels too low level for me for this change

cascheberg replied:
Do you have another suggestion?
I think I remember that handling it inside `recipient` was much more difficult. This solution has the nice effect that "you" is always at the beginning of the string. Or is it something else that you would like to change?

cascheberg commented:
Another straight forward solution would be to remove your own number from the list of contacts that where added to the group and change the wording to "User1, User2 also joined the group."

Seeing your own entryg as just a phone number is really not great.

altring commented:
When will we see this?

alan-signal commented:
![Thank you!](https://media0.giphy.com/media/l3q2FnW3yZRJVZH2g/giphy.gif)

Unfortunately we will not merge this as this issue has since been resolved. Groups now list "Me".

cascheberg commented:
@alan-signal the issue still exists. It is about the group update message that is shown when a group is created. I created #8888 to document it.
This PR is quite old and I did not care to maintain it because apparently no one was interest in merging it, but it's nice to see that community PRs are getting some attention again.

alan-signal commented:
I miss-understood, I'll reopen and give it a closer look.

cascheberg commented:
@alan-signal Thanks. As I understood from Moxie this change was too low-level for a rather small improvement. Let me know what you think. I can rebase or rework it.

alan-signal commented:
Thanks but I think it might have been fixed, will close again when it makes it to public.

Mark5595 commented:
Hi there, 

Seems like this issue still exists for group MMS messages. It appears in 3 views that my own number is displayed as a raw phone number rather than a more helpful `me`, `you` or even nothing. 

1. Conversation list (main screen on app reset) view where you can jump into a conversation shows my number in a group MMS conversation.

2. Clicking into a group MMS conversation shows my number at the header / title bar 

3. From a group MMS clicking the header (next to the back button) opening up the the chat settings view shows my number 

The one place where my raw number is not exposed is when clicking on the people icon from a group MMS screen there the `Group Members` modal opens and appropriately shows `me` instead of my raw number.   

Displaying my number takes up valuable screen real estate where I would rather see my peers in the conversation as I know I am in a conversation that I can see (otherwise it wouldn't show up).

cascheberg commented:
@alan-signal I noticed that there have been changes regarding PROFILE_DISPLAY, but the issue described and fixed here still remains in the current release. I made a rebase on the current master, please check again if this is still useful. I added screenshots to the original description. The commit is much simpler now than it was three years ago, because some required changes have already been implemented lately.
